mlflow/utils/environment.py

```diff
@@ -666,6 +667,29 @@ def _deduplicate_requirements(requirements):
     return [str(req) for req in deduped_reqs.values()]
 
 
+def _remove_incompatible_requirements(requirements: list[str]):
+    def fetch_requirement_name(req: str) -> str:
+        try:
+            return Requirement(req).name
+        except Exception:
+            return req
+
+    req_name = {fetch_requirement_name(req) for req in requirements}
+    if "databricks-connect" in req_name and any(
```

Comment:
```suggestion
    req_names = {fetch_requirement_name(req) for req in requirements}
    if "databricks-connect" in req_names and any(
```

it's a collection. Let's make it plural.